src/libraries/System.Private.CoreLib/src/System.Private.CoreLib.Shared.projitems

```diff
@@ -891,6 +891,7 @@
     <Compile Include="$(MSBuildThisFileDirectory)System\Runtime\CompilerServices\StateMachineAttribute.cs" />
     <Compile Include="$(MSBuildThisFileDirectory)System\Runtime\CompilerServices\StringFreezingAttribute.cs" />
     <Compile Include="$(MSBuildThisFileDirectory)System\Runtime\CompilerServices\StrongBox.cs" />
+    <Compile Include="$(MSBuildThisFileDirectory)System\Runtime\CompilerServices\SuppressByRefLikeConstraintChecksAttribute.cs" />
```

Comment:
Just want to confirm that the compiler needs these annotations in CoreLib implementation only and it does not need them in the reference assembly. Is that correct?

Reply:
> Just want to confirm that the compiler needs these annotations in CoreLib implementation only and it does not need them in the reference assembly. Is that correct?

That depends how you build the core library. Since the attribute is internal, no one else can really use it. Therefore, if we are able to build the core library, we should be good, I think.

Reply:
The reference assemblies are built by Roslyn too, from different sources. I think we need this attribute in the reference assemblies too. Otherwise, the Span/ReadOnlySpan sources in reference assemblies will fail to build:

https://github.com/dotnet/runtime/blob/main/src/libraries/System.Runtime/ref/System.Runtime.cs#L5151-L5159